waspc/src/Wasp/AppSpec/JSON.hs

```diff
@@ -3,18 +3,30 @@
 module Wasp.AppSpec.JSON
   ( JSON (..),
     emptyObject,
+    nullValue,
   )
 where
 
 import qualified Data.Aeson as Aeson
-import qualified Data.ByteString.Lazy.UTF8 as ByteStringLazyUTF8
+import qualified Data.Aeson.Text as Aeson.Text
+import qualified Data.Aeson.Types as Aeson.Types
 import Data.Data (Data)
+import qualified Data.Text.Lazy as TextL
 
 newtype JSON = JSON Aeson.Value
   deriving (Eq, Data)
 
 instance Show JSON where
-  show (JSON val) = ByteStringLazyUTF8.toString $ Aeson.encode val
+  show (JSON val) = TextL.unpack . Aeson.Text.encodeToLazyText $ val
+
+instance Aeson.ToJSON JSON where
+  toJSON (JSON val) = val
+
+instance Aeson.FromJSON JSON where
+  parseJSON val = return $ JSON val
```

Comment:
Yeah, you can derive Generic there and just leave off the `where` part. I was thinking here it may be preferable to show/remind explicitly that these really are in fact `Aeson` values. But we can go the Generic route 👍🏻

Reply:
Ok, that sounds like a good reason, I was just asking!